packages/web/src/components/RewardForm.js

```diff
@@ -0,0 +1,122 @@
+// Module imports
+import {
+	useCallback,
+	useEffect,
+    useState,
+} from 'react'
+import PropTypes from 'prop-types'
+import { useRouter } from 'next/router'
+
+
+// Local imports
+import { Field } from 'components/Field'
+import { Form } from 'components/Form'
+import { FormButton } from 'components/FormButton'
+import { useAuth } from 'contexts/AuthContext'
+import API from 'helpers/API'
+
+
+
+function RewardForm(props) {
+	const { campaign, showModal } = props
+	const { user } = useAuth()
+	const Router = useRouter()
+
+
+	  const createReward = useCallback(async (reward, campaign) => {
+        reward.uid = campaign.ownerID
+        reward.campaignID = campaign.id
+		const response = await API.post({
+			body: reward,
+			route: '/rewards/new',
+		})
+		const responseJSON = await response.json()
+		return responseJSON.id
+	}, [])
+
+	const handleSubmit = useCallback(async formState => {
+		const {
+			isValid,
+			values,
+		} = formState
+
+		if (isValid) {
+			try {
+				const newRewardID = await createReward(values, campaign)
+                showModal(false);
+			} catch(error) {
+				alert(`Unexpected error saving reward: ` + error)
+			}
+		}
+	}, [])
+	return (
+		<Form
+			initialValues={{
+				title: '',
+				cost: '',
+                maxRedemptions: '',
+				perUser: false,
+                cooldown: '',
+				color: '',
+			}}
+			onSubmit={handleSubmit}>
+			<Field
+				id="title"
+				isRequired
+				label="Name"
+				type="text"
+				maxLength={25}
+                title="Enter a name for Twitch reward" />
+
+			<Field
+				id="cost"
+                isRequired
+				label="Cost"
+				type="number"
+                title="Enter channel points cost of reward"/>
+            <div className="redemptionFields">
```

Comment:
Does this `div` do anything?

Reply:
I was envisioning using a class here to add styling so that these fields were inline/float with each other on the same line (which makes sense to me from a UX perspective since they are related). But I abandoned that. Can add it back in later.

Reply:
Legit. Bulma also has a utility for doing that sort of thing: https://bulma.io/documentation/form/general/#form-group